posthog/models/filters/filter.py

```diff
@@ -72,49 +260,11 @@ def __init__(self, data: Optional[Dict[str, Any]] = None, request: Optional[Http
             }
         elif not data:
             raise ValueError("You need to define either a data dict or a request")
+
+        self._data = data
         self._date_from = data.get(DATE_FROM)
         self._date_to = data.get(DATE_TO)
-        self.entities = data.get(ENTITIES, [])
         self.properties = self._parse_properties(data.get(PROPERTIES))
-        self.selector = data.get(SELECTOR, [])
-        self.interval = data.get(INTERVAL)
-        self.display = data.get(DISPLAY)
-        self.selector = data.get(SELECTOR)
-        self.shown_as = data.get(SHOWN_AS)
-        self.breakdown = self._parse_breakdown(data)
-        self.breakdown_type = data.get(BREAKDOWN_TYPE)
-        self.breakdown_value = data.get(BREAKDOWN_VALUE)
-        self._compare = data.get(COMPARE, "false")
-        self.insight = data.get(INSIGHT)
-        self.session_type = data.get(SESSION)
-        self.path_type = data.get(PATH_TYPE)
-        self.start_point = data.get(START_POINT)
-        self._offset = data.get(OFFSET)
-
-        if data.get(ACTIONS):
-            self.entities.extend(
-                [Entity({**entity, "type": TREND_FILTER_TYPE_ACTIONS}) for entity in data.get(ACTIONS, [])]
-            )
-        if data.get(EVENTS):
-            self.entities.extend(
-                [Entity({**entity, "type": TREND_FILTER_TYPE_EVENTS}) for entity in data.get(EVENTS, [])]
-            )
-        self.entities = sorted(self.entities, key=lambda entity: entity.order if entity.order else -1)
-
-    def _parse_breakdown(self, data: Dict[str, Any]) -> Optional[Union[str, List[Union[str, int]]]]:
-        breakdown = data.get(BREAKDOWN)
-        if not isinstance(breakdown, str):
-            return breakdown
-        try:
-            return json.loads(breakdown)
-        except (TypeError, json.decoder.JSONDecodeError):
-            return breakdown
-
-    def _parse_target_entity(self, target_entity_data) -> Optional[Entity]:
-        if target_entity_data:
-            data = json.loads(target_entity_data)
-            return Entity({"id": data["id"], "type": data["type"]})
-        return None
 
     def to_dict(self) -> Dict[str, Any]:
```

Comment:
Note - we can probably automatically generate this method as well if we go the mixin route. :)

Reply:
```python
    def to_dict(self) -> Dict[str, Any]:
        ret = {}

        for key, value in self.__dict__.items():
            if key in ['entities'] or key.startswith('_'):
                continue
            if isinstance(value, list) and len(value) == 0:
                continue
            if not isinstance(value, list) and not value:
                continue
            if not isinstance(value, (list, bool, int, float, str)) and not hasattr(value, '__dict__'):
                continue
            if isinstance(value, Entity):
                value = value.to_dict()
            if isinstance(value, list) and isinstance(value[0], Entity):
                value = [entity.to_dict() for entity in value]
            ret[key] = value

        return ret
```